scripts/deploy.sh

```diff
@@ -1,60 +1,11 @@
 #!/bin/bash
-source "$(dirname "${BASH_SOURCE}")/lib/init.sh"
 
-# from makefile
-BROKER_IMAGE=$1
-REGISTRY=$2
-DOCKERHUB_ORG=$3
+BROKER_ROOT=$(dirname "${BASH_SOURCE}")/..
 
-# can be overridden via my_local_dev_vars
-PROJECT=${ASB_PROJECT}
-ASB_SCHEME="https"
-ROUTING_SUFFIX="172.17.0.1.nip.io"
-OPENSHIFT_TARGET="https://kubernetes.default"
-REGISTRY_TYPE="dockerhub"
-DEV_BROKER="true"
-LAUNCH_APB_ON_BIND="false"
-OUTPUT_REQUEST="true"
-RECOVERY="true"
-REFRESH_INTERVAL="600s"
-SANDBOX_ROLE="edit"
-VARS="${VARS:=''}"
-
-# load development variables
-asb::load_vars
-
-# check the variables that do not have defaults
-asb::validate_var "BROKER_IMAGE" $BROKER_IMAGE
-asb::validate_var "REGISTRY" $REGISTRY
-asb::validate_var "DOCKERHUB_USERNAME" $DOCKERHUB_USERNAME
-asb::validate_var "DOCKERHUB_PASSWORD" $DOCKERHUB_PASSWORD
-asb::validate_var "DOCKERHUB_ORG" $DOCKERHUB_ORG
-asb::validate_var "REFRESH_INTERVAL" $REFRESH_INTERVAL
-
-VARS+=" -p BROKER_IMAGE=${BROKER_IMAGE} \
-  -p ASB_SCHEME=${ASB_SCHEME} \
-  -p ROUTING_SUFFIX=${ROUTING_SUFFIX} \
-  -p OPENSHIFT_TARGET=${OPENSHIFT_TARGET} \
-  -p DOCKERHUB_ORG=${DOCKERHUB_ORG} \
-  -p DOCKERHUB_PASS=${DOCKERHUB_PASS} \
-  -p DOCKERHUB_USER=${DOCKERHUB_USER} \
-  -p REGISTRY_TYPE=${REGISTRY_TYPE} \
-  -p REGISTRY_URL=${REGISTRY} \
-  -p DEV_BROKER=${DEV_BROKER} \
-  -p LAUNCH_APB_ON_BIND=${LAUNCH_APB_ON_BIND} \
-  -p OUTPUT_REQUEST=${OUTPUT_REQUEST} \
-  -p RECOVERY=${RECOVERY} \
-  -p REFRESH_INTERVAL=${REFRESH_INTERVAL} \
-  -p SANDBOX_ROLE=${SANDBOX_ROLE}"
-
-# cleanup old deployment
-asb::delete_project ${PROJECT}
-
-# delete the clusterrolebinding to avoid template error
-oc delete clusterrolebindings --ignore-not-found=true asb
-
-# deploy
-oc new-project ${PROJECT}
-oc process -f ${BROKER_TEMPLATE} \
-  -n ${PROJECT} \
-  ${VARS} | oc create -f -
+if [[ "${KUBERNETES}" ]]; then
+    echo "Using Kubernetes"
+    "${BROKER_ROOT}/scripts/kubernetes/deploy.sh"
```

Comment:
My recommendation is to bring in the `scripts/kubernetes/deploy.sh` and `scripts/openshift/deploy.sh` into this file.

Reply:
What I didn't want to do is turn these scripts into if k8s then do 100 things else openshift do these 100 things. With two separate files there's a clean separation because these are two separate scripts. So anything kubernetes related I know exists in ```scripts/kubernetes/...```.  It's not the best situation, but Bash isn't exactly the best for creating abstraction.

Reply:
@rthallisey +1 keep them separate. They will be easier to manage. Unless they are 100% the same, otherwise it will turn into a bunch of `if kube else` etc.